Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/data/RemoteDataSource.kt

```diff
@@ -0,0 +1,160 @@
+package com.mentalmachines.droidcon_boston.data
+
+import com.mentalmachines.droidcon_boston.R
+import com.mentalmachines.droidcon_boston.modal.Media
+import com.mentalmachines.droidcon_boston.modal.QuotedTweet
+import com.mentalmachines.droidcon_boston.modal.TweetWithMedia
+import com.mentalmachines.droidcon_boston.utils.toDate
+import com.twitter.sdk.android.core.TwitterCore
+import com.twitter.sdk.android.core.models.Tweet
+import com.mentalmachines.droidcon_boston.modal.Tweet as ViewTweet
+import java.text.SimpleDateFormat
+import java.util.Collections
+import java.util.Locale
+
+class RemoteDataSource : DataSource {
+
+    companion object {
+
+        private const val TWEET_COUNT = 100
+
+        private lateinit var dataSource: RemoteDataSource
+
+        fun getInstance(): RemoteDataSource {
+            if (!::dataSource.isInitialized) {
+                dataSource = RemoteDataSource()
+            }
+            return dataSource
+        }
+    }
+
+    override fun getTweets(): List<TweetWithMedia> {
+        val response = TwitterCore
+            .getInstance()
+            .guestApiClient
+            .searchService
+            .tweets(
+                "%23DroidconBos",
+                null,
+                null,
+                null,
+                "latest",
+                TWEET_COUNT,
+                null,
+                0,
+                0,
+                true
+            ).execute()
+        return mapTweets(response.body()?.tweets ?: Collections.emptyList())
+    }
+
+    private fun mapTweets(tweets: List<Tweet>): List<TweetWithMedia> {
```

Comment:
Any way we can break out this method? It gets nested really deep and can be hard to keep track of the context.

Reply:
Done